src/psyclone/domain/lfric/transformations/lfric_raise_call_2_invoke_trans.py

```diff
@@ -68,39 +67,38 @@ def apply(self, call, index, options=None):
 
         call_name = None
         calls = []
+        table = call.scope.symbol_table
+
         for idx, call_arg in enumerate(call.children):
 
-            arg_info = []
             if call.argument_names[idx]:
                 call_name = f"'{call_arg.value}'"
             elif isinstance(call_arg, ArrayReference):
                 # kernel or builtin misrepresented as ArrayReference
                 args = call_arg.pop_all_children()
-                name = call_arg.name
-                if name in builtins:
-                    node_type = LFRicBuiltinFunctor
-                    type_symbol = call.scope.symbol_table.lookup(name)
+                if call_arg.name in BUILTIN_FUNCTOR_MAP:
+                    calls.append(BUILTIN_FUNCTOR_MAP[call_arg.name].create(
```

Comment:
It would be good if we would do some refactoring here:
add a new base class `LFRicFunctor`, based on `KernelFunctor`, and it then being the base class for both `LFRicBuiltinFunctor` and `LFRicKernelFunctor`. And `LFRicFunctor` would then hide the difference between builtins and kernel calls by providing a simple `create` function:`calls.append(LFRicFunctor.create(....)` (taking all required arguments for builtin and kernel call).

Not sure if `LFRicFunctor` is the best name (since it's based on `KernelFunctor` - my suggestion: rename `KernelFunctor` to `Functor` - I think that's much clearer, since it's odd that KernelFunctor is a base class for LFRicBuiltinFunctor).